tools/toolchains/__init__.py

```diff
@@ -1111,6 +1092,42 @@ def get_config_header(self):
         self.config_processed = True
         return self.config_file
 
+    @staticmethod
+    def generic_check_executable(tool_key, executable_name, levels_up,
+                                 nested_dir=None):
+        """
+        Positional args:
+        tool_key: the key to index TOOLCHAIN_PATHS
+        executable_name: the toolchain's named executable (ex. armcc)
+        nested_dir: the directory within TOOLCHAIN_PATHS where the executable
```

Comment:
Ah, whoops. Changed it to accomodate gcc_arm and forgot to move it to a kw arg docstring.